src/Libraries/RevitNodesUI/Elements.cs

```diff
@@ -215,37 +210,46 @@ public ElementsInView()
             OutPortData.Add(new PortData("elements", Properties.Resources.PortDataAllVisibleElementsToolTip));
             RegisterAllPorts();
 
-            DocumentManager.Instance.CurrentUIApplication.ViewActivated +=
-                RevitDynamoModel_RevitDocumentChanged;
+            DynamoRevit.AddIdleAction(
+                () =>
+                {
+                    DocumentManager.Instance.CurrentUIApplication.ViewActivated +=
+                        RevitDynamoModel_RevitDocumentChanged;
 
-            DocumentManager.Instance.CurrentUIApplication.Application.DocumentOpened +=
-                RevitDynamoModel_RevitDocumentChanged;
+                    DocumentManager.Instance.CurrentUIApplication.Application.DocumentOpened +=
+                        RevitDynamoModel_RevitDocumentChanged;
 
-            RevitServicesUpdater.Instance.ElementsDeleted +=
-                RevitServicesUpdaterOnElementsDeleted;
-            RevitServicesUpdater.Instance.ElementsModified +=
-                RevitServicesUpdaterOnElementsModified;
-            RevitServicesUpdater.Instance.ElementsAdded +=
-                RevitServicesUpdaterOnElementsAdded;
+                    RevitServicesUpdater.Instance.ElementsDeleted +=
+                        RevitServicesUpdaterOnElementsDeleted;
+                    RevitServicesUpdater.Instance.ElementsModified +=
+                        RevitServicesUpdaterOnElementsModified;
+                    RevitServicesUpdater.Instance.ElementsAdded +=
+                        RevitServicesUpdaterOnElementsAdded;
 
-            RevitDynamoModel_RevitDocumentChanged(null, null);
+                    RevitDynamoModel_RevitDocumentChanged(null, null);
+                });
         }
 
         public override void Dispose()
         {
+            DynamoRevit.AddIdleAction(
+                () =>
+                {
+                    DocumentManager.Instance.CurrentUIApplication.ViewActivated -=
+                        RevitDynamoModel_RevitDocumentChanged;
+
+                    DocumentManager.Instance.CurrentUIApplication.Application.DocumentOpened -=
+                        RevitDynamoModel_RevitDocumentChanged;
+
+                    RevitServicesUpdater.Instance.ElementsDeleted -=
+                        RevitServicesUpdaterOnElementsDeleted;
+                    RevitServicesUpdater.Instance.ElementsModified -=
+                        RevitServicesUpdaterOnElementsModified;
+                    RevitServicesUpdater.Instance.ElementsAdded -=
+                        RevitServicesUpdaterOnElementsAdded;
+                });
+
             base.Dispose();
```

Comment:
It should be fine I guess but now these event handlers are unregistered way after `base.Dispose` method is called, if for some reasons these events are raised then it's important to ensure they don't access things that are already disposed of.